File: Maximum Matrix Sum.py
# https://leetcode.com/problems/maximum-matrix-sum/?envType=daily-question&envId=2024-11-24
# Maximum Matrix Sum

class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        
        n = len(matrix)
        total_sum = 0
        min_abs_value = float('inf')
        negative_count = 0

        for i in range(n):
            for j in range(n):
                value = matrix[i][j]
                total_sum += abs(value)
                min_abs_value = min(min_abs_value, abs(value))
                if value < 0:
                    negative_count += 1

        if negative_count % 2 != 0:
            total_sum -= 2 * min_abs_value

        return total_sum

File: Maximum Matrix Sum.js
// https://leetcode.com/problems/maximum-matrix-sum/?envType=daily-question&envId=2026-01-05
// Maximum Matrix Sum

/**
 * @param {number[][]} matrix
 * @return {number}
 */
var maxMatrixSum = function(matrix) {

    const n = matrix.length;
    let totalSum = 0;
    let minAbsValue = Infinity;
    let negativeCount = 0;

    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            const value = matrix[i][j];
            const absValue = Math.abs(value);

            totalSum += absValue;
            minAbsValue = Math.min(minAbsValue, absValue);

            if (value < 0) {
                negativeCount++;
            }
        }
    }

    if (negativeCount % 2 !== 0) {
        totalSum -= 2 * minAbsValue;
    }

    return totalSum;
};